lib/Validation.groovy

```diff
@@ -0,0 +1,115 @@
+import groovy.json.JsonSlurper
+import groovy.util.logging.Log
+
+class Validation {
+
+    /*
+    * Function to loop over all parameters defined in schema and check
+    * whether the given paremeters adhere to the specificiations
+    */
+    private static void validateParameters(params, json_schema, log, workflow){
+
+        def json = new File(json_schema).text
+        def Map json_params = (Map) new JsonSlurper().parseText(json).get('definitions')
+        def specified_param_keys = params.keySet()
+        def nf_params = ['profile', 'config', 'c', 'C', 'syslog', 'd', 'dockerize', 
+                        'bg', 'h', 'log', 'quiet', 'q', 'v', 'version']
+        def valid_params = []
+        def expected_params = []
+        def blacklist  = ['hostnames'] // ignored parameters
```

Comment:
Why are you skipping `hostnames`?

Also, a thread going around the internet a while back was that we should try to avoid adding in variable names like this due to `#BlackLivesMatter` awareness 🤔

Reply:
Oh shoot ...okay will change it.

Hmm blacklisted it for now because it parses as a LinkedHashMap. Will look into it.